src/index.js

```diff
@@ -157,6 +161,13 @@ function Swarm (peerInfo) {
           this.muxedConns[pi.id.toB58String()] = {}
           this.muxedConns[pi.id.toB58String()].muxer = muxedConn
           this.muxedConns[pi.id.toB58String()].conn = conn // to be able to extract addrs
+
+          self.emit('peer-mux-established', pi)
+
+          muxedConn.on('close', () => {
+            delete self.muxedConns[pi.id.toB58String()]
+            self.emit('peer-mux-closed', pi)
```

Comment:
We could. The thing I'm trying to distinguish, is that there is 'conns' and 'muxedConns' and for bitswap, it only really matters the 'muxedConns'. Saying 'peer-disconnect' and not emitting it for a regular conn exit could mislead a user.